test/test_example_parsers/test_dice_roll_parser.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+
+from plusminus.examples.example_parsers import DiceRollParser
+
+
+@pytest.fixture
+def die():
+    return DiceRollParser()
+
+
+class TestSingleROll:
+    def test_single_roll_max_one(self, die):
+        assert die.evaluate('d1') == 1
+
+    @pytest.mark.parametrize('die_max',
+                             [2, 4, 5, 10, 100, ],
+                             )
+    def test_single_roll_ranges(self, die,
+                                die_max):
+        assert 1 <= die.evaluate(f'd{die_max}') <= die_max
+
+    def test_single_roll_error_empty_range(self, die):
+        with pytest.raises(ValueError):
+            die.evaluate('d0')
+
+
+class TestMultipleRolls:
+    @pytest.mark.parametrize('die_max',
+                             [2, 3, 4, 5, 10, 100, ]
+                             )
+    def test_zero_rolls_returns_zero(self, die,
+                                     die_max):
+        assert die.evaluate(f'0d{die_max}') == 0
+
+    @pytest.mark.parametrize('num_rolls',
+                             [1, 2, 4, 5, 6, 10, 100, ],
+                             )
+    def test_multiple_rolls_max_one(self, die,
+                                    num_rolls):
+        assert num_rolls <= die.evaluate(f'{num_rolls}d1') == num_rolls
+
+    @pytest.mark.parametrize('num_rolls',
+                             [1, 2, 4, 5, 10, 100, ],
+                             )
+    @pytest.mark.parametrize('die_max',
+                             [2, 4, 5, 10, 100, ]
+                             )
```

Comment:
You can usefully stack parametrizations, so can pass (1,2) and (3,4) and get (1,3), (1,4), (2,3), (2,4).
I just found this out tonight, but I like it very much!